testing/ldc/APKBUILD

```diff
@@ -0,0 +1,110 @@
+# The main package depends on them.
+# A user might want to install `runtime` when they have
+# a dynamically-linked D program.
+pkgname="ldc"
+pkgver="1.8.0"
+pkgrel=0
+pkgdesc="The LLVM-based D Compiler"
+url="https://github.com/ldc-developers/ldc"
+arch="x86_64"
+license="BSD-3-Clause; Some parts BSL-1.0; Some parts Artistic-1.0 or GPL-2.0-or-later; Some parts NCSA; Some parts NCSA and MIT"
+depends="libexecinfo"
+makedepends="chrpath cmake llvm5-libs libexecinfo-static"
+install=""
+subpackages="$pkgname-runtime $pkgname-doc $pkgname-static"
+provides="dlang-compiler"
+source="https://github.com/ldc-developers/ldc/releases/download/v1.8.0/ldc2-1.8.0-alpine-linux-x86_64.tar.xz
+		ldc2.package.conf"
+builddir="$srcdir/"
+
+build() {
+	rootdir="${srcdir}/ldc2-1.8.0-alpine-linux-x86_64/"
+
+	# Unfortunately the upstream-provided version has the builder's /home/ in its rpath - strip it
+	chrpath -d "${rootdir}/bin/ldc2"
+	chrpath -d "${rootdir}/bin/ldmd2"
+	# This exe is not installed, just used in build, but do it for completeness
+	chrpath -d "${rootdir}/bin/ldc-build-runtime"
+
+	# Now build the runtime - the compiler itself is already built by upstream
+	"${rootdir}"/bin/ldc-build-runtime --buildDir="${builddir}/ldc-runtime-build/"
+
+	# CMake added the rpaths to the shared libs - strip them
+	chrpath -d "${builddir}"/ldc-runtime-build/lib/*.so
+	:
+}
+
+check() {
+	rootdir="${srcdir}/ldc2-1.8.0-alpine-linux-x86_64/"
+	libroot="${builddir}/ldc-runtime-build/lib/"
+
+	echo 'import std.stdio; void main () { try throw new Exception(null); catch (Exception) writeln("Hello World!"); }' > hello_world.d
+	if [ "$(${rootdir}/bin/ldmd2 -conf='' -I${rootdir}/import/ldc -I${rootdir}/import/ -L-L${libroot} -defaultlib=phobos2-ldc,druntime-ldc -L-lexecinfo -L-rpath=${libroot} -run hello_world.d)" != "Hello World!" ]; then
+		return 1
+	fi
+	:
+}
+
+package() {
+	depends="${pkgname}-runtime ${pkgname}-static"
+	rootdir="${srcdir}/ldc2-1.8.0-alpine-linux-x86_64/"
+
+	install -s -D "${rootdir}/bin/ldc2"	 "${pkgdir}/usr/bin/ldc2"
+	install -s -D "${rootdir}/bin/ldmd2" "${pkgdir}/usr/bin/ldmd2"
+
+	install -D "${srcdir}/ldc2.package.conf" "${pkgdir}/etc/ldc2.conf"
+
+	install -D "${rootdir}/LICENSE" "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
+
+	# Install the import file
+	for impfile in $(find "${rootdir}/import" -name "*.d*");
+	do
+		target="${impfile##*import/}"
+		install -D "$impfile" "${pkgdir}/usr/include/dlang/${pkgname}-${pkgver}/${target}"
+	done
+
+	# Install the static runtime libraries
+	for libn in "libdruntime-ldc" "libdruntime-ldc-debug" "libphobos2-ldc" "libphobos2-ldc-debug";
+	do
+		install -D "${builddir}/ldc-runtime-build/lib/${libn}.a" "${pkgdir}/usr/lib/${libn}.a"
+	done
+
+	# Install the shared runtime libraries
+	for libn in "libdruntime" "libphobos2";
+	do
+		install -D "${builddir}/ldc-runtime-build/lib/${libn}-ldc-shared.so.2.0.78" "${pkgdir}/usr/lib/${libn}-ldc-shared.so.2.0.78"
+	done
+}
```

Comment:
No need for the curly braces about all of these. Please install `atools` and run apkbuild-lint which will automatically find such errors for you.

Reply:
Removed curly braces everywhere.
`apkbuild-lint` didn't warn me about it, but did point two other issues (pkgver & pkgname shouldn't be quoted). I also had to change some comments, as it complained about my usage of backtips in comments.

Side note: Shouldn't that be part of the CI ?